include/net/net_ip.h

```diff
@@ -40,18 +40,21 @@ extern "C" {
 #define PF_UNSPEC	0	/* Unspecified.  */
 #define PF_INET		2	/* IP protocol family.  */
 #define PF_INET6	10	/* IP version 6.  */
+#define PF_CAN		13	/* CAN protocol family */
 
 /** Address families.  */
 #define AF_UNSPEC	PF_UNSPEC
 #define AF_INET		PF_INET
 #define AF_INET6	PF_INET6
+#define AF_CAN		PF_CAN	/* CAN address family */
 
 /** Protocol numbers from IANA */
 enum net_ip_protocol {
 	IPPROTO_ICMP = 1,
 	IPPROTO_TCP = 6,
 	IPPROTO_UDP = 17,
 	IPPROTO_ICMPV6 = 58,
+	CAN_RAW = 252,
```

Comment:
"The SocketCAN concept extends the Berkeley sockets API in Linux by introducing a new protocol family, PF_CAN, that coexists with other protocol families like PF_INET for the Internet Protocol. The communication with the CAN bus is therefore done analogously to the use of the Internet Protocol via sockets."
Source: https://en.wikipedia.org/wiki/SocketCAN

Reply:
ok

Reply:
Few comments here:

1. I'm relatively fine with this particular change, the problem can be seen as in enum naming.
2. BUT, linux has it as a separate define in can.h, and I would like to extend my suggestion is not to invent our own special ways and follow well-known/established API examples (and belonging of symbols to headers is part of API in C).
3. "The communication with the CAN bus is therefore done analogously to the use of the Internet Protocol via sockets." - "Analogously" does not mean "the same". Because otherwise, absolutely all protocols work analogously - shoving bits back and forth. But somehow, this patch patches the IP protocol files and not I2C or MQTT.  Looking a bit down the same Wikipedia article, example shows that what is sent over CAN sockets is `struct can_frame`. This is very, very different from normal TCP or UDP sockets.